upf/upf_cli.c

```diff
@@ -285,15 +286,27 @@ upf_nat_pool_add_del_command_fn (vlib_main_t * vm,
 	}
     }
 
-  nwi_name = upf_name_to_labels (nwi_s);
-  vec_free (nwi_s);
+  /*
+   * Extra port range check here because port values are parsed into
+   * u32 instead of u16
+   */
+  if (min_port < UPF_NAT_MIN_PORT ||
+      max_port > UPF_NAT_MAX_PORT || min_port > max_port)
+    error = clib_error_return (0, "Invalid port range");
```

Comment:
I would have use `goto done` here instead of the else block, but that is only a aesthetic thing.